lib/rouge/lexers/solidity.rb

```diff
@@ -0,0 +1,184 @@
+# -*- coding: utf-8 -*- #
+
+module Rouge
+  module Lexers
+    class Solidity < RegexLexer
+      tag 'solidity'
+      filenames '*.sol', '*.solidity'
+      mimetypes 'text/x-solidity'
+
+      title "Solidity"
+      desc "Solidity, an Ethereum smart contract programming language"
+
+      # optional comment or whitespace
+      ws = %r((?:\s|//.*?\n|/[*].*?[*]/)+)
+      id = /[a-zA-Z_][a-zA-Z0-9_]*/
+
+
+      def self.analyze_text(text)
+        return 1 if text.shebang? 'pragma solidity'
+      end
+
+      # TODO: seperate by "type"
+      def self.keywords
+        @keywords ||= Set.new %w(
+          anonymous as assembly break constant continue contract do delete
+          else enum event external for function hex if indexed interface
+          internal import is library mapping memory modifier new payable
+          public pragma private return returns storage struct throw
```

Comment:
there's a new keyword `pure` now too.

Reply:
Don't forget `view`, as well.

Reply:
I could go through my latest update to the Pygments lexer, and fish out the rest of the changes.

Reply:
@veox would be nice to make this up to date with Solidity 0.6.0 and/or with the linguist code

Reply:
These and most others are now added.